### PR TITLE
HostId with Docker/K8S

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: nservicebus/hosting/override-hostid.md [hunk 1]
```diff
@@ -25,3 +25,10 @@ There are scenarios where the rules used by NServiceBus to generate a `hostid` a
 Manual configuration is required when deployments may end up in different paths than previously deployed versions (e.g. using [Octopus Deploy](https://octopus.com/)). The `hostid` must remain the same across restarts unless the physical host has changed.
 
 snippet: HostIdFixer
+
+
+## Stable host identifiers when using Docker
```
Comment (andreasohlund) [inline nservicebus/hosting/override-hostid.md:30]: Do we need to mention this here https://docs.particular.net/monitoring/metrics/install-plugin#configuration-instance-id  as well?

File: nservicebus/hosting/override-hostid.md [hunk 2]
```diff
@@ -25,3 +25,10 @@ There are scenarios where the rules used by NServiceBus to generate a `hostid` a
...
+
+Docker containers hosted in Kubernetes pose unique challenges to creating a stable host identifier, as a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) will not have a stable, unique identifier. This results in the endpoint having a new, random identity in ServicePulse and ServiceInsight every time the [Pod]() containing the endpoint is (re)scheduled on a different node.
```
Comment (ramonsmits) [inline nservicebus/hosting/override-hostid.md:32]: Pod does not have a link.

File: nservicebus/hosting/override-hostid.md (hunk 1 above)
Comment (ramonsmits) [reply nservicebus/hosting/override-hostid.md:30]: @andreasohlund it copies the ID from core.

**Comment by DavidBoike** [comment]
Thanks for the catch on the link @ramonsmits!

> Good addition to the doco although I think we would be better of with a page dedicated to docker and its nuances. Point to pages like these for customizations.

Agree but that exercise would be much larger. This is the scope I can get away with from a weekly doco review.

**Pull request merged**